solr/core/src/java/org/apache/solr/cloud/Overseer.java

```diff
@@ -303,7 +303,7 @@ private void checkIfIamStillLeader() {
       String path = OVERSEER_ELECT + "/leader";
       byte[] data;
       try {
-        data = zkClient.getData(path, null, stat, true);
+        data = zkClient.getData(path, null, stat, false);
```

Comment:
i changed the `retryOnConnLoss` to false here because `checkIfIamStillLeader` is the teardown hook.  The retry was causing tests to fail due to the com.carrotsearch.randomizedtesting.ThreadLeakControl check seeing threads that were waiting on retry, and also, it seems a little weird to do this retrying on a teardown hook anyway.